### PR TITLE
rxm: Add MSG capability

### PR DESCRIPTION
**Pull request by a-ilango**

### CODE REVIEW

File: include/fi_util.h [hunk 1]
```diff
@@ -292,6 +293,62 @@ int ofi_get_src_addr(uint32_t addr_format,
...
+	struct util_cmap_handle **handles;
+	struct util_cmap_keypool *keypool;
+	struct dlist_entry peer_list;
+	ofi_cmap_free_handle_func free_handle;
+	fastlock_t lock;
+};
+
+void ofi_cmap_update_state(struct util_cmap_handle *handle,
+		enum util_cmap_state state);
+int ofi_cmap_add_handle_addr(struct util_cmap *cmap, struct util_cmap_handle *handle,
```
Comment (j-xiong) [inline include/fi_util.h]: The names of this and some of the following functions are somewhat confusing: it's not easy to tell the real differences between similarly named functions; the "get" function does deletion. If these functions are supposed to be part of the "util" API exposed to the provider writers, please try to refine the names to be more descriptive. Also try to minimize the set of functions need to be exposed. For those functions that require a lock being held, please also add that information to the header file.

Comment (a-ilango) [inline include/fi_util.h]: The two similarly named functions take in different arguments to add a handle. They could be merged with the presence/absence of the args defining the behavior. That would reduce the number of functions exported. Also, ofi_cmap_del_handles is not supposed to be exported. I shall remove it from the header.

> the "get" function does deletion

This function moves the handle from "peer list" to handle array in cmap which appears like deletion. An alternate approach would be to put the move part in a separate function which should be called when a provider does an av_insert.
> For those functions that require a lock being held, please also add that information to the header file.

Sure, will do. Thank you for reviewing.

**Comment by a-ilango** [comment]
@j-xiong I have updated the PR. Merged the add_handle functions. I'll modify the get_handle function as part of a later PR since that involves more changes.

**Comment by j-xiong** [comment]
👍

**Pull request merged**